pynamodb/connection/transactions.py

```diff
@@ -0,0 +1,124 @@
+from pynamodb.exceptions import GetError
+
+from pynamodb.constants import (
+    RETURN_VALUES_ON_CONDITION_FAILURE, ITEM, RETURN_VALUES, RESPONSES,
+    TRANSACTION_CONDITION_CHECK_REQUEST_PARAMETERS, TRANSACTION_DELETE_REQUEST_PARAMETERS,
+    TRANSACTION_GET_REQUEST_PARAMETERS, TRANSACTION_PUT_REQUEST_PARAMETERS, TRANSACTION_UPDATE_REQUEST_PARAMETERS
+)
+from pynamodb.models import _ModelFuture
+
+
+class Transaction(object):
+
+    _connection = None
+    _hashed_models = None
+    _results = None
+    _return_consumed_capacity = None
+
+    def __init__(self, connection, return_consumed_capacity=None):
+        self._connection = connection
+        self._hashed_models = set()
+        self._return_consumed_capacity = return_consumed_capacity
+
+    def _commit(self):
+        raise NotImplementedError()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._commit()
```

Comment:
We commit even when there was an exception?

Reply:
was mostly copying what happens in batches: https://github.com/pynamodb/PynamoDB/blob/master/pynamodb/models.py#L99

Reply:
updated to only commit if there are no exception args passed in